### DISCUSSION TITLE
External Camera and Sound via Sound Card

### DISCUSSION THREAD

**Discussion by cortnum** [HaishinKit/HaishinKit.swift#1127] [Idea]
Yo, first of all, great framework. It has helped me built our app LiveLive.

Second of all, I was wondering if the framework already support getting audio from external microphones and cameras added via apple Camera Adapter. I am interested in streaming audio from a sound card so that users can have more control over and mix the audio before it enters the stream.

Once again, great framework and thanks for the hard work!

**Comment by shogo4405** [comment]
If you can get  an *External Camera* with  AVCaptureDevice.devices() method.
Then you can use an External Camera.
